tests/unit/test_execution_environment_utils.py

```diff
@@ -0,0 +1,292 @@
+# (c) 2012-2019, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+import logging
+import os
+import pytest
+
+from galaxy_importer.utils import execution_environment as ee_utils
+
+
+logger = logging.getLogger(__name__)
+
+
+EE_YAML = """---
+version: 1
+dependencies:
+  galaxy: requirements.yml
+  python: requirements.txt
+  system: bindep.txt
+
+additional_build_steps:
+  prepend: |
+    RUN whoami
+    RUN cat /etc/os-release
+  append:
+    - RUN echo This is a post-install command!
+    - RUN ls -la /etc
+"""
+
+
+EE_DIR_YAML = """---
+version: 1
+dependencies:
+  galaxy: meta/requirements.yml
+  python: meta/requirements.txt
+  system: meta/bindep.txt
+
+additional_build_steps:
+  prepend: |
+    RUN whoami
+    RUN cat /etc/os-release
+  append:
+    - RUN echo This is a post-install command!
+    - RUN ls -la /etc
+"""
+
+REQUIREMENTS_YML = """
+---
+collections:
+  - name: https://github.com/AlanCoding/awx.git#awx_collection,ee_req
+    type: git
+  - name: https://github.com/AlanCoding/azure.git
+    version: ee_req
+    type: git
+"""
+
+
+IMPORTED_REQUIREMENTS_YML = {
+    'collections': [
+        {
+            'name': 'https://github.com/AlanCoding/awx.git#awx_collection,ee_req',
+            'type': 'git'
+        },
+        {
+            'name': 'https://github.com/AlanCoding/azure.git',
+            'version': 'ee_req',
+            'type': 'git'
+        },
+    ]
+}
+
+
+REQUIREMENTS_TXT = """
+ansible
+paramiko
+"""
+
+BINDEP_TXT = """
+gcc
+python3-devel
+libcurl-devel
+openssl-devel
+libxml2-devel
```

Comment:
minor nit - it's expected to have the bindep data on it like `libxml2-devel [platform:rpm]`, but at the stage of processing you're doing, it won't do anything with that, so it would get that entry but then return it as-is.

Reply:
I didn't come across any `bindep.txt` files with the platform data.  I'll track down an example and update my tests.